code/controllers/subsystem/SSshuttles.dm

```diff
@@ -479,6 +479,41 @@ SUBSYSTEM_DEF(shuttle)
 	custom_escape_shuttle_loading = FALSE
 	return loaded_shuttle
 
+/datum/controller/subsystem/shuttle/proc/set_trader_shuttle(datum/map_template/shuttle/trader/template)
+	var/obj/docking_port/mobile/trader/trade_shuttle = getShuttle("trader")
+	if(trade_shuttle)
+		var/obj/docking_port/stationary/docked_id = trade_shuttle.get_docked()
+		if(docked_id.id != "trader_base")
```

Comment:
```suggestion
		if(docked_id?.id != "trader_base")
```
One more small thing, just in case the trade shuttle is somehow not at any dock at all (even a transit dock).